### DISCUSSION TITLE
What's the recommended way for installing Firefox on AL2023?

### DISCUSSION THREAD

**Discussion by trivikr** [amazonlinux/amazon-linux-2022#418] [Q&A]
When I search for how to install Firefox on Amazon Linux, the [top result](https://stackoverflow.com/a/72052840) is to install dependencies like gtk3 and libdbusmenu, before installing firefox from tar.

When attempted in AL2023, the dependencies are nor available and firefox installation throws "Couldn't load XPCOM" error.

```console
$ export BROWSERS_SRC_DIR="/usr/src/browsers" && mkdir -p $BROWSERS_SRC_DIR

$ curl -L --output $BROWSERS_SRC_DIR/firefox-latest.tar.bz2 \
    "https://download.mozilla.org/?product=firefox-latest&os=linux64&lang=en-US"

$ tar xvjf $BROWSERS_SRC_DIR/firefox-latest.tar.bz2 -C /usr/local

$ ln -s /usr/local/firefox/firefox /usr/bin/firefox

$ /usr/bin/firefox --version
XPCOMGlueLoad error for file /usr/local/firefox/libxul.so:
libdbus-glib-1.so.2: cannot open shared object file: No such file or directory
Couldn't load XPCOM.

$ yum install -y libdbus-glib-1-2
...
No match for argument: libdbus-glib-1-2
```

```console
$ yum install -y gtk3
...
Package gtk3-3.24.30-4.amzn2023.0.4.x86_64 is already installed.

$ yum install -y libdbusmenu
...
Error: Unable to find a match: libdbusmenu
```

What's the recommended way for installing Firefox on AL2023?

**Comment by trivikr** [comment]
I created request for missing packages `libdbusmenu` and `libdbusmenu-jsonloader` in https://github.com/amazonlinux/amazon-linux-2023/issues/425

**Comment by matejsp** [comment]
Firefox just released nigtly release of linux arm build. I haven't tried it yet but looks promising.
https://blog.nightly.mozilla.org/2024/04/19/firefox-nightly-now-available-for-linux-on-arm64/
